### DISCUSSION TITLE
Warning ao fazer "muitas" requests para uma api - Nextjs

### DISCUSSION THREAD

**Discussion by CDInacio** [frontendbr/forum#2384] [Perguntas e Respostas]
Eu preciso fazer 3 ou mais requests para uma api chamada jikan (cujos dados retornados serão responsaveis por popular a rota principal ("/")), só que eu estou recebendo o seguinte warning:

`Warning: data for page "/" is 198 kB which exceeds the threshold of 128 kB, this amount of data can reduce performance.`

Gostaria de saber se tem uma forma mais correta de fazer essas chamadas de api. Qual seria a melhor altrernativa?

Segue o código:
```
export default function Home({ topAnime, currentSeasonAnimes }) {
  return (
    <>
      <Hero />
      <Container>
        <Top text="Top animes" topData={topAnime} />
        {/* ... */}
      </Container>
    </>
  );
}

export const getStaticProps = async () => {
  const topAnime = await api.get("/top/anime?filter=airing&limit=5");
  const currentSeasonAnimes = await api.get("/seasons/now?limit=5");
  const recomendations = await api.get("/recommendations/anime");

  return {
    props: {
      topAnime: topAnime.data.data,
      currentSeasonAnimes: currentSeasonAnimes.data.data,
      recomendations: recomendations.data.data,
    },
  };
};
```

**Comment by ninetails** [comment]
Uma dica daria pra jogar as 3 chamadas num [Promise.allsettled](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) pra já chamar tudo em paralelo.

Sobre o warning, parece ser algo sobre o tamanho dos dados.

Se você tiver acesso à API, diria pra mandar pro front somente dados que serão exibidos na tela ou computados no teu app, enxugando a quantidade de dados que vc envia pro front.

Se você não tiver acesso à API, tente fazer isso após a própria chamada e antes de retornar as props.

No teu exemplo, você precisa mesmo da chamada de `recommendations` já que não está sendo utilizada na `<Home />`?

Você também pode quebrar a requisição para deixar o essencial pro SEO vindo do `getStaticProps` e dados que não forem críticos pra SEO, deixar numa requisição na própria página usando [alguma estratégia](https://17.reactjs.org/docs/concurrent-mode-suspense.html#traditional-approaches-vs-suspense), mas aí vc vai jogar a responsabilidade da requisição pro usuário e não mais pro servidor. Não seria muito indicado se o servidor não for público ou não for algo pensado pra escalar e aguentar tais requisições.
